oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/KeycloakRBACAuthorizer.java

```diff
@@ -474,16 +507,21 @@ private boolean denyIfTokenInvalid(BearerTokenWithPayload token) {
     private JsonNode handleFetchingGrants(BearerTokenWithPayload token) {
         // Fetch authorization grants
         JsonNode grants = null;
+
         try {
             grants = fetchAuthorizationGrants(token.value());
             if (grants == null) {
                 grants = JSONUtil.newObjectNode();
             }
-        } catch (HttpException e) {
-            if (e.getStatus() == 403) {
-                grants = JSONUtil.newObjectNode();
+        } catch (Throwable e) {
+            if (e instanceof HttpException) {
```

Comment:
If we're only interested in handling `HttpException`, let's catch `Exception` instead of `Throwable`